File: src/ios/App/Models/Address.m
// ==============================================================================
//
// This file is part of the WelHair
//
// Create by Welfony <support@welfony.com>
// Copyright (c) 2013-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================

#import "Address.h"
#import "CityManager.h"

@implementation Address

- (Address *)initWithDic:(NSDictionary *)dictionary
{
    self = [super init];
    if (self) {
        self.id = [[dictionary objectForKey:@"AddressId"] intValue];
        self.userName = [dictionary objectForKey:@"ShippingName"];
        self.phoneNumber = [dictionary objectForKey:@"Mobile"];
        self.detailAddress = [dictionary objectForKey:@"Address"];

        if ([dictionary objectForKey:@"IsDefault"]) {
            self.city = [[CityManager SharedInstance] getCityById:[[dictionary objectForKey:@"City"] intValue]];
        }

        if ([dictionary objectForKey:@"IsDefault"]) {
            self.isDefault = [[dictionary objectForKey:@"IsDefault"] intValue] > 0;
        }
    }

    return self;
}

- (NSDictionary *)dictionaryFromAddress:(Address *)address
{
    NSMutableDictionary *dic = [[NSMutableDictionary alloc] initWithCapacity:4];
    [dic setObject:[NSNumber numberWithInt:address.id] forKey:@"AddressId"];

    return dic;
}

@end

File: src/www/api/routes/address.php
<?php

// ==============================================================================
//
// This file is part of the WelStory.
//
// Create by Welfony Support <support@welfony.com>
// Copyright (c) 2012-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================

use Welfony\Controller\API\AddressController;

$app->get('/users/:userId/addresses', function ($userId) use ($app) {
    $ctrl = new AddressController();
    $ctrl->listByUser($userId);
})->conditions(array('userId' => '\d{1,10}'));

$app->post('/addresses', function () use ($app) {
    $ctrl = new AddressController();
    $ctrl->create();
});

$app->put('/addresses/:addressId', function ($addressId) use ($app) {
    $ctrl = new AddressController();
    $ctrl->update($addressId);
})->conditions(array('addressId' => '\d{1,10}'));

$app->post('/addresses/:addressId/remove', function ($addressId) use ($app) {
    $ctrl = new AddressController();
    $ctrl->remove($addressId);
})->conditions(array('addressId' => '\d{1,10}'));

$app->post('/addresses/:addressId/default', function ($addressId) use ($app) {
    $ctrl = new AddressController();
    $ctrl->setToDefault($addressId);
})->conditions(array('addressId' => '\d{1,10}'));
